File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.java

import java.util.*;
class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        
        Map<Character, Integer> typeMap = new HashMap<>();	
		
		for(int i = 0; i < survey.length; i++) {
			if(choices[i] > 4) {
				typeMap.put(survey[i].charAt(1), typeMap.getOrDefault(survey[i].charAt(1), 0) + (choices[i] -               4));
			} else if(choices[i] < 4){
				typeMap.put(survey[i].charAt(0), typeMap.getOrDefault(survey[i].charAt(0), 0) + (4 - choices                 [i]));
			}
		}
		
		
		answer += typeMap.getOrDefault('R', 0) < typeMap.getOrDefault('T', 0) ? 'T' : 'R';
		answer += typeMap.getOrDefault('C', 0) < typeMap.getOrDefault('F', 0) ? 'F' : 'C';
		answer += typeMap.getOrDefault('J', 0) < typeMap.getOrDefault('M', 0) ? 'M' : 'J';
		answer += typeMap.getOrDefault('A', 0) < typeMap.getOrDefault('N', 0) ? 'N' : 'A';
        
        return answer;
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var answer = '';
    let personality = ["R", "T", "C", "F", "J", "M", "A", "N"];
    let scoreMap = [[0, 0], [0, 3], [0 ,2], [0, 1], [0, 0] , [1, 1], [1, 2], [1, 3]];
    let map = new Map();
    for (p of personality) map.set(p, 0);
    for (let i = 0; i < survey.length; i++){
        let choice = scoreMap[choices[i]][0];
        let score = scoreMap[choices[i]][1];
        map.set(survey[i][choice] ,map.get(survey[i][choice]) + score);
    }
    answer += map.get("T") > map.get("R") ? 'T': 'R';
    answer += map.get("F") > map.get("C") ? 'F': 'C';
    answer += map.get("M") > map.get("J") ? 'M': 'J';
    answer += map.get("N") > map.get("A") ? 'N': 'A';
    return answer;
}